openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/GlossaryTermRepository.java

```diff
@@ -225,6 +228,35 @@ protected void updateTags(String fqn, String fieldName, List<TagLabel> origTags,
       }
     }
 
+    private void updateChildName(EntityReference child, String parentFqn) throws IOException {
+      GlossaryTerm childTerm = get(null, child.getId(), new Fields(List.of("parent", "children")));
+      String oldFqn = childTerm.getFullyQualifiedName();
+      EntityReference parent = childTerm.getParent();
+      parent.setFullyQualifiedName(parentFqn);
+      setFullyQualifiedName(childTerm);
+      String newFqn = childTerm.getFullyQualifiedName();
+      dao.update(childTerm.getId(), JsonUtils.pojoToJson(childTerm));
+      // update all the tag usages
+      daoCollection.tagUsageDAO().updateTagFqn(oldFqn, newFqn);
```

Comment:
@vivekratnavel any reason to update tags?

Reply:
When the name of a GlossaryTerm gets updated, its FQN gets updated. If we don't update tagUsage, then all the assets that were tagged with this GlossaryTerm will no longer be tagged and the records in tagUsage table become invalid and dangling pointers.